Add Rocky Linux to with_yum

lbakman opened the pull request:
Changelog: Bugfix: Add Rocky Linux to with_yum.
Docs: omit

Similar to: #9460, only for Rocky Linux.

- [x] Refer to the issue that supports this Pull Request.
- [x] If the issue has missing info, explain the purpose/use case/pain/need that covers this Pull Request.
- [x] I've read the [Contributing guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [x] I've followed the PEP8 style guides for Python code.
- [ ] I've opened another PR in the Conan docs repo to the ``develop`` branch, documenting this one. 

<sup>**Note:** By default this PR will skip the slower tests and will use a limited set of python versions. Check [here](https://github.com/conan-io/conan/blob/develop/.github/PR_INCREASE_TESTING.md) how to increase the testing level by writing some tags in the current PR body text.</sup>

lbakman committed:
Add Rocky Linux to with_yum

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/conan-io/conan?pullRequest=11212) <br/>All committers have signed the CLA.

memsharded reviewed (changes requested):
Thanks for contributing this @lbakman 

But this place you are doing the change is deprecated in favor of the new PackageManager implementation, so at least you should include changes in https://github.com/conan-io/conan/blob/develop/conan/tools/system/package_manager.py too. 

An entry check to an existing unit test, like ``test_package_manager_distro`` would be good too.

lbakman committed:
Add Rocky Linux to package_manager_test.py and added entry check to test_package_manager_distro.

lbakman commented:
Good points.

I have included the changes to package_manager.py and added rocky/yum as a parameter to test_package_manager_distro.

One thing that caught my attention: I can see that both "rhel" and "centos" has been moved to "dnf". Even though I added "rocky" as a "yum" mapping, I would argue that both "almalinux" and "rocky" should be moved to "dnf" as well. This would be a breaking change compared to "conans/client/tools/oss.py", but I guess the same goes for "rhel" and "centos".

memsharded commented:
Thanks!
